### PR TITLE
refactor: cleanup of `TransactionalColumnFamily` 

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

Since #8793 grew quite large and a bit unwieldy, this PR just contains the cleanup of `TransactionalColumnFamily`:
- removes unnecessary type parameters in methods
- removes unnecessary parameters such as `context`
- public methods use `ensureInOpenTransaction`, private methods don't
- simplifies the list of methods and the call chain between them

I've tried to optimize for readabilty and consistency in how the public methods of `TransactionalColumnFamily` are implemented.
The tradeoff is a small amount of duplicated code, for example in the two public `forEach` methods.
I think the code duplication is "good": at least for me, it makes the code more readable because I only have to know how `forEachInPrefix` works to then understand what the different `forEach`, `whileEqualPrefix` and `whileTrue` methods do.

### CODE REVIEW

File: zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java [hunk 1]
```diff
@@ -47,20 +47,12 @@
     columnFamilyContext = new ColumnFamilyContext(columnFamily.ordinal());
   }
 
-  private void ensureInOpenTransaction(
```
Comment (pihme) [inline zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java:50]: :x: You came up with some code conventions, e.g. that `ensureInOpenTransaction` is only called from public methods, not from private methods. Please document these conventions either on class level or on the method level. 

Same for the preference to make `forEachInPrefix` the central method to route to

**Review by pihme — APPROVED** [review]
Yeah, I like it a lot. The diff still is confusing to look at, but the final version of the class looks much better. Also, you ere able to eliminate 20% of the lines even with adding more code duplication.

Please document the conventions you came up with.

If you want me to look at it again, I am happy to. But you can also go ahead and merge it right away.

File: zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java (hunk 1 above)
Comment (oleschoenburg) [reply zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java:50]: :+1: I added some documentation on both the class and method level.

**Comment by oleschoenburg** [comment]
bors retry

**Comment by pihme** [comment]
bors retry

**Pull request merged**